nep-3.mediawiki

```diff
@@ -0,0 +1,137 @@
+
+<pre>
+{
+  "hash": "0x562851057d8afbc08fabc8c438d7cc771aef2195",
+  "entrypoint": "main",
+  "functions": [],
+  "events": []
+}
+</pre>
+
+<code>hash</code> is the script hash of the contract. It is encoded as a hexadecimal string in big-endian.
+
+<code>entrypoint</code> indicates that which is the entry point of the contract in <code>functions</code>.
+
+<code>functions</code> is an array of Function objects which describe the details of each function in the contract.
+
+<code>events</code> is an array of Event objects which describe the details of each event in the contract.
+
+===Function===
+
+Function object has the following structure:
+
+<pre>
+{
+  "name": "transfer",
+  "parameters": [],
+  "returntype": "Boolean"
+}
+</pre>
+
+<code>name</code> is the name of the function, which can be any valid identifier.
+
+<code>parameters</code> is an array of Parameter objects which describe the details of each parameter in the function.
+
+<code>returntype</code> indicates the return type of the function. It can be one of the following values: <code>Signature</code>, <code>Boolean</code>, <code>Integer</code>, <code>Hash160</code>, <code>Hash256</code>, <code>ByteArray</code>, <code>PublicKey</code>, <code>String</code>, <code>Array</code>, <code>InteropInterface</code>, <code>Void</code>.
+
+===Event===
+
+Event object has the following structure:
+
+<pre>
+{
+  "name": "refund",
+  "parameters": []
+}
+</pre>
+
+<code>name</code> is the name of the event, which can be any valid identifier.
+
+<code>parameters</code> is an array of Parameter objects which describe the details of each parameter in the event.
+
+===Parameter===
+
+Parameter object has the following structure:
+
+<pre>
+{
+  "name": "from",
+  "type": "Hash160"
+}
+</pre>
+
+<code>name</code> is the name of the parameter, which can be any valid identifier.
+
+<code>type</code> indicates the type of the parameter. It can be one of the following values: <code>Signature</code>, <code>Boolean</code>, <code>Integer</code>, <code>Hash160</code>, <code>Hash256</code>, <code>ByteArray</code>, <code>PublicKey</code>, <code>String</code>, <code>Array</code>, <code>InteropInterface</code>.
+
+===ParameterType===
+
+ParameterType enum has the following values:
+
+{|
+!name
+!description
+|-
+| Signature
+| A signature of a transaction or block which is generated by the user.
+|-
+| Boolean
+| A boolean value can be either <code>true</code> or <code>false</code>.
+|-
+| Integer
+| An arbitrarily large integer whose value in theory has no upper or lower bounds.
+|-
+| Hash160
+| A 160-bits integer.
+|-
+| Hash256
+| A 256-bits integer.
+|-
+| ByteArray
+| A byte array.
+|-
+| PublicKey
+| An ECC public key which is encoded with compressed mode.
+|-
+| String
+| A string which is encoded in UTF-8.
+|-
+| Array
+| An array of objects.
```

Comment:
Could you elaborate on what "object" means in this context? It doesn't seem to be defined as a standard type